docs/apis-tools/frontend-development/01-task-applications/02-user-task-lifecycle.md

```diff
@@ -0,0 +1,148 @@
+
+Our best practices, as implemented in the Tasklist web application, are as follows:
+
+- `update` and `complete` operations can only be performed by the assigned user or an admin/manager.
+- Users can only see tasks to which they are assigned and tasks to which a user group of which they are a member is assigned as a candidate.
+- When a task is returned to the queue (i.e. the assignee is cleared), its data and status are reset to "open".
+- Only admins/managers can reassign tasks.
+- Task agents can return tasks, but must provide a comment as to why they are doing so.
+- Task agents can mark tasks with a follow-up date. These then disappear from their individual task list until the follow-up date is reached. The `open` status is preserved, or the task is moved to the `paused` status if it has already been processed. The task remains assigned to the user.
+
+The following image demonstrates how this is similarly implemented in the Camunda Tasklist web application:
+
+<img src={TaskLifecycleUI} style={{width: 600}} alt="Task lifecycle and assignment in the Camunda Tasklist UI" />
+
+Make sure that you create your own validation logic that matches your use case.
+
+## Implement the lifecycle with the task API
+
+<!-- TODO update wih links to API explorer once available -->
+
+To implement task lifecycle operations with the task API, you have to call the respective endpoints:
+
+- `POST /user-tasks/:taskKey/assignment` or `DELETE /user-tasks/:taskKey/assignee` to change task assignment.
+- `PATCH /user-tasks/:taskKey/update` to update a task.
+- `POST /user-tasks/:taskKey/completion` to complete a task.
+
+All these endpoints (except `DELETE`) allow to send a custom `action` attribute via the payload. The `action` attribute carries any arbitrary string and can be used to track any lifecycle event, including those mentioned above.
+
+#### `POST /user-tasks/:taskKey/assignment`
+
+Use the `assignment` endpoint to change the task assignment. Use the `action` attribute to indicate the cause of the change, including `claim`, `reassign`, or `assign`.
+
+#### `PATCH /user-tasks/:taskKey/update`
+
+Use the `update` endpoint to change candidate users, groups, the due date, or the follow-up date, by defining the `changeSet`. You can also send it with an empty `changeSet` and just pass an `action`. Use it to send `start`, `pause` and `resume` actions. In addition, you can send anything of interest or relevant for the audit log such as `escalate`, `requestFurtherInformation`, `uploadDocument`, or `openExternalApp`.
```

Comment:
🔧 The attribute is `changeset` (see [OpenAPI def](https://github.com/camunda/zeebe/blob/d062f533f7dde333074b0d16664311d041345a53/zeebe/gateway-protocol/src/main/proto/rest-api.yaml#L290)).